### PR TITLE
Refactor/powerbi

### PR DESCRIPTION
**Pull request by rickpr**

### CODE REVIEW

**Comment by rickpr** [comment]
Not sure what exactly to do about the `mypy` errors . Looking into it.

**Comment by Mr0grog** [comment]
Ah, I’ll bet you’re getting errors like:

```
covid19_sfbayarea/data/san_mateo/power_bi_querier.py:1: error: Cannot find implementation or library stub for module named 'covid19_sfbayarea.data.power_bi.querier'
```

Because you aren’t using relative imports, and we don’t actually install the module in CI (because we don’t have a real `setup.py`):

https://github.com/sfbrigade/data-covid19-sfbayarea/blob/7b1876f0caa851e6a6e53f02cb31a2e5d174d0d6/.circleci/config.yml#L12-L22

**Comment by rickpr** [comment]
Restricted this to sharing just the most useful parts an figured out the mypy errors. Might follow up with more stuff, but I think this is good for one PR.

**Comment by rickpr** [comment]
So it turned out the folder I was importing from needed a `__init__.py` - that solved the `Cannot find implementation or library stub for module`

File: covid19_sfbayarea/data/power_bi/querier.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    def __init__(self) -> None:
+        self.function = getattr(self, 'function', 'CountNotNull')
+        self._assert_required_attributes_are_set()
+
+    def get_data(self) -> Union[List, Dict]:
+        response_json = self._fetch_data()
+        return self._parse_data(response_json)
+
+    def _fetch_data(self) -> Dict:
+        response = post(self.base_uri, headers = { 'X-PowerBI-ResourceKey': self.powerbi_resource_key }, json = self._query_params())
```
Comment (Mr0grog) [inline covid19_sfbayarea/data/power_bi/querier.py:27]: Would be nice to reformat this across multiple lines for readability.

**Review by Mr0grog — CHANGES_REQUESTED** [review]

File: covid19_sfbayarea/data/power_bi/querier.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        return pairs
+
+    # PowerBI uses the key 'R' to represent repeated values.
+    # The values to repeat are indexed by bits, starting with 1. These bits are sent as decimal.
+    # So element 0 has a value of 1, element 1 has a value of 2, element 2 has a value of 4 and they keep doubling.
+    # These values are then added together.
+    # For example, 14 would mean that the repeated indexes 1, 2, and 3 (2nd, 3rd, and 4th elements) repeat.
+    def _determine_repeated_values(self, r: int) -> List[int]:
+        r_in_binary = reversed('{:b}'.format(r))
+        return [ bool(int(one_or_zero)) for one_or_zero in r_in_binary ]
```
[the remark this answers is not in the document]
Comment (Mr0grog) [reply covid19_sfbayarea/data/power_bi/querier.py:138]: Fixed a typo in comment above, if following by e-mail.

[the remark this answers is not in the document]
Comment (rickpr) [reply covid19_sfbayarea/data/power_bi/querier.py:138]: Thanks for the great explanation and cool simplification of this!

**Pull request closed without merging**